### PR TITLE
LDAP fixes and enhancements

### PR DESCRIPTION
**Pull request by tbugfinder**
Add LDAP debug and library check.
Add some ActiveDirectory specific features.

### CODE REVIEW

**Comment by dlancia** [comment]
Please merge.  I have manually made these changes/fixes on my iLibrarian install as well and having them in the main branch would be great so I dont have to modify iLibrarian after every release.

**Comment by mkucej** [comment]
Sure, I will merge this when a new version will be worked on.

File: ilibrarian-default.ini [hunk 1]
```diff
@@ -95,20 +95,35 @@ top_window_color = "FFFFFF"
...
 
-; LDAP server port
+;Global Catalog queries
+;ldap_server = "ldap://ldap.forumsys.com:3268"
+;ldap_server = "ldaps://ldap.forumsys.com:3269"
 
-ldap_port = 389
+; LDAP server port (at least not used on Windows) using LDAP URI
+
+;ldap_port = 389
```
Comment (mkucej) [inline ilibrarian-default.ini]: Must be set to empty string.

File: ilibrarian-default.ini [hunk 2]
```diff
@@ -14,7 +14,7 @@ temp_path = ""
 
 greeting = "I, Librarian"
 
-[openlinks]
+[openlinks]b
```
Comment (mkucej) [inline ilibrarian-default.ini]: Extra character.

File: ilibrarian-default.ini [hunk 3]
```diff
@@ -117,7 +132,9 @@ ldap_basedn = "dc=example,dc=com"
 ; Proxy user to use for username lookups (without base DN)
 
 ;ldap_binduser_rdn = "cn=read-only-admin"
-ldap_binduser_rdn = ""
+;ldap_binduser_rdn = ""
```
Comment (mkucej) [inline ilibrarian-default.ini]: Line 135 not necessary.

File: ilibrarian-default.ini [hunk 4]
```diff
@@ -117,7 +132,9 @@ ldap_basedn = "dc=example,dc=com"
 ; Proxy user to use for username lookups (without base DN)
 
 ;ldap_binduser_rdn = "cn=read-only-admin"
```
Comment (mkucej) [inline ilibrarian-default.ini]: Move this line below description.

**Comment by mkucej** [comment]
Thank you much for your contribution!

**Pull request merged**